server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -167,9 +168,7 @@
  * }
  * </pre>
  */
-public abstract class BlobStoreRepository extends AbstractLifecycleComponent implements Repository {
-
-    private BlobContainer snapshotsBlobContainer;
+public abstract class BlobStoreRepository<BS extends BlobStore> extends AbstractLifecycleComponent implements Repository {
```

Comment:
Do we really need a generic here? It looks to me that it's useful for innerBlobStore() / blobContainer() methods used in tests but otherwise it's not really required.